io.openems.edge.evcs.goe.chargerhome/src/io/openems/edge/evcs/goe/chargerhome/GoeChargerHomeImpl.java

```diff
@@ -0,0 +1,340 @@
+				this.channel(GoeChannelId.STATUS_GOE).setNextValue(json.get("car").getAsInt());			
+				this.channel(Evcs.ChannelId.STATUS).setNextValue(ConvertGoeStatus(json.get("car").getAsInt()));				
+				this.channel(GoeChannelId.VOLTAGE_L1).setNextValue(nrg.get(0).getAsInt());
+				this.channel(GoeChannelId.VOLTAGE_L2).setNextValue(nrg.get(1).getAsInt());
+				this.channel(GoeChannelId.VOLTAGE_L3).setNextValue(nrg.get(2).getAsInt());
+				this.channel(GoeChannelId.CURRENT_L1).setNextValue(nrg.get(4).getAsInt()*100);
+				this.channel(GoeChannelId.CURRENT_L2).setNextValue(nrg.get(5).getAsInt()*100);
+				this.channel(GoeChannelId.CURRENT_L3).setNextValue(nrg.get(6).getAsInt()*100);
+				this.channel(GoeChannelId.ACTUAL_POWER).setNextValue(nrg.get(11).getAsInt()*10);
+		
+				this.channel(Evcs.ChannelId.PHASES).setNextValue(phases);				
+				this.channel(Evcs.ChannelId.MAXIMUM_POWER).setNextValue(GetMaxPower(phases));	
+				this.channel(Evcs.ChannelId.MINIMUM_HARDWARE_POWER).setNextValue(GetMinHardwarePower(phases, cabelCurrent));	
+				this.channel(Evcs.ChannelId.MAXIMUM_HARDWARE_POWER).setNextValue(GetMaxHardwarePower(cabelCurrent));	
+				this.channel(Evcs.ChannelId.CHARGE_POWER).setNextValue(nrg.get(11).getAsInt()*10);
+				this.channel(Evcs.ChannelId.CHARGING_TYPE).setNextValue(ChargingType.AC);
+				
+				this.channel(GoeChannelId.ENERGY_TOTAL).setNextValue(json.get("eto").getAsInt()*100);				
+				this.channel(Evcs.ChannelId.ENERGY_SESSION).setNextValue(json.get("dws").getAsInt()*10/3600);
+			
+				this.channel(GoeChannelId.ERROR).setNextValue(err);
+				this.channel(Evcs.ChannelId.CHARGINGSTATION_COMMUNICATION_FAILED).setNextValue(false);
+				
+				this.setEnergySession();
+				this.setPower();				
+				
+			}
+			else{
+				this.channel(Evcs.ChannelId.CHARGINGSTATION_COMMUNICATION_FAILED).setNextValue(true);
+			}
+			
+			break;
+		}
+	}
+	
+	private int ConvertGoeStatus(int status) {
+		switch(status) {
+		case 1: // ready for charging, car unplugged
+			return Status.NOT_READY_FOR_CHARGING.getValue();
+		case 2: // charging
+			return Status.CHARGING.getValue();
+		case 3: // waiting for car
+			return Status.READY_FOR_CHARGING.getValue(); 
+		case 4: // charging finished, car plugged
+			return Status.CHARGING_FINISHED.getValue();
+		default:
+			return Status.UNDEFINED.getValue();
+		}
+	}
+	
+	private int ConvertGoePhase(int phase) {
+		int Phasen = (byte)phase & 0b00111000;			
+		switch(Phasen) {
+		case 8: // 0b00001000: Phase 1 is active
+			return 1;
+		case 24: // 0b00011000: Phase 1+2 is active
+			return 2;
+		case 56: // 0b00111000: Phase1-3 are active
+			return 3;
+		default:
+			return 0;
+		}
+	}
+	
+	private int GetMaxHardwarePower(int cableCurrent) {
+		int MaxPower = 0;
+		int phases = 3;
+		if (cableCurrent < this.MaxCurrent && cableCurrent > 0) {
+			MaxPower = phases*cableCurrent*230/1000;
+		}
+		else {
+			MaxPower = phases*this.MaxCurrent*230/1000;
+		}
+		return MaxPower;
```

Comment:
Here you could include that is considered how much power the inverter can. Because that would then be the maximum power that can be used without having to draw power from the grid (e.g. GoodWe GW 10 --> 10kW max so the maximum power without having to draw from the grid would be 10kW).